frontend/src/features/properties/components/forms/subforms/PagedBuildingForms.tsx

```diff
@@ -166,7 +168,7 @@ const PagedBuildingForms: React.FC<PagedBuildingFormsProps> = (props: PagedBuild
                 ))}
               </Form.Control>
             </Form.Group>
-            {!props.disabled && !!props.allowEdit && (
+            {!props.disabled && (props.isAdmin || !buildings[currentPage]?.id) && (
```

Comment:
only show building deletes to admins or when the current building hasn't been saved (no id).